.github/workflows/build.yaml

```diff
@@ -0,0 +1,279 @@
+  component_descriptor:
+    runs-on: ubuntu-latest
+    container:
+      image: python:alpine
+    permissions:
+      contents: write
+      id-token: write
+    needs:
+      - params
+      - version
+      - build
+    steps:
+      - name: install gardener-ocm
+        run: |
+          pip install \
+            gardener-oci \
+            gardener-ocm \
+      - name: merge ocm-resource-fragments
+        uses: actions/upload-artifact/merge@v4
+        with:
+          name: ocm-oci-resources
+          # only merge ocm-fragments emitted from ocm-oci-build-action
+          pattern: ocm-oci-build*
+          separate-directories: true
+          delete-merged: true
+      - name: download-ocm-oci-resources
+        uses: actions/download-artifact@v4
+        with:
+          name: ocm-oci-resources
+          path: ocm-oci-resources.d
+
+      - name: generate OCM Component-Descriptor
+        run: |
+          env | sort
+          version=${{ needs.version.outputs.effective_version }}
+          ocm_repo=${{ needs.params.outputs.ocm_repository }}
+
+          echo "base component-descriptor"
+          gardener-ocm \
+            create \
+            --name github.com/${{ github.repository }} \
+            --version  ${version} \
+            --ocm-repo ${ocm_repo} \
+            --provider sap-se \
+            --label '{
+              "name": "cloud.gardener.cnudie/responsibles",
+              "value":  [
+                {
+                  "type": "githubTeam",
+                  "teamname": "gardener/ci-maintainers",
+                  "github_hostname": "github.com"
+                }
+              ]
+            }' \
+            > component-descriptor.yaml
+            cat component-descriptor.yaml
+
+            echo "adding main-source"
+            echo "\
+            name: main-source
+            version: ${version}
+            type: git
+            labels:
+              - name: cloud.gardener/cicd/source
+                value:
+                  repository-classification: main
+            access:
+              type: github
+              repoUrl: github.com/${{ github.repository }}
+              version: ${version}
+              commit: ${{ github.sha }}
+              ref: ${{ github.ref }}
+            " \
+            | gardener-ocm \
+              append source \
+              --file component-descriptor.yaml
+
+            for rf in $(find ocm-oci-resources.d -type f -name ocm-resources.yaml); do
+              echo "adding ocm-resources from ${rf}"
+              cat ${rf} | gardener-ocm \
+                append resource \
+                --file component-descriptor.yaml
+            done
+
+          echo "Component-Descriptor ready to be uploaded:"
+          cat component-descriptor.yaml
+      - name: authenticate
+        if: ${{ inputs.release }}
+        uses: gardener/cc-utils/.github/actions/gar-auth@master
+      - name: upload OCM Component-Descriptor
+        if: ${{ inputs.release }}
```

Comment:
Why should the component descriptor only be uploaded in case it's a "release" run? What about snapshot component descriptors?

Reply:
@8R0WNI3 : in case of _this_ particular pipeline, I think we have no need for snapshot-component-descriptors.

I plan to refactor this (specific) workflow into a reusable workflow, though. the latter should of course feature an option to configure whether to upload component-descriptor.

Admittedly, we might also decide to upload snapshot-component-descriptors for this pipeline, of course